### ISSUE TITLE
可以补充context-gating到那个即插即用里面吗？

### ISSUE THREAD

**Issue by 1187697147** [pprp/SimpleCVReproduction#6]

**Comment by pprp** [comment]
Learnable pooling with Context Gating for video classification 这篇文章？

**Comment by 1187697147** [comment]
对对，我是刚入门的，觉得你这个写的很好用。



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "PJDong"<notifications@github.com&gt;; 
发送时间: 2020年6月19日(星期五) 下午4:36
收件人: "pprp/SimpleCVReproduction"<SimpleCVReproduction@noreply.github.com&gt;; 
抄送: "　"<1187697147@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [pprp/SimpleCVReproduction] 可以补充context-gating到那个即插即用里面吗？ (#6)





 
Learnable pooling with Context Gating for video classification 这篇文章？

**Comment by 1187697147** [comment]
这篇文章的pytorch写的。
class GatingContext(nn.Module):
    def __init__(self, dim, add_batch_norm=True):
        super(GatingContext, self).__init__()
        self.dim = dim
        self.add_batch_norm = add_batch_norm
        self.gating_weights = nn.Parameter(
            nn.init.normal_(
                torch.empty(dim, dim), mean=0, std=(1 / math.sqrt(dim)),
            )
        )
        self.sigmoid = nn.Sigmoid()
        if add_batch_norm:
            self.gating_biases = None
            self.batch_norm = nn.BatchNorm1d(dim)
        else:
            self.gating_biases = nn.Parameter(
                torch.nn.init.normal_(
                    torch.empty(dim), mean=0, std=(1 / math.sqrt(dim)),
                )
            )
            self.batch_norm = None

    def forward(self, x):
        gates = torch.matmul(x, self.gating_weights)
        if self.add_batch_norm:
            gates = self.batch_norm(gates)
        else:
            gates = gates + self.gating_biases
        gates = self.sigmoid(gates)
        activation = x * gates
        return activation

**Comment by pprp** [comment]
已加入，欢迎补充新的模块
